framework/src/base/MooseApp.C

```diff
@@ -636,10 +636,13 @@ MooseApp::setupOptions()
     Moose::registerExecFlags(_factory);
 
   // Print the header, this is as early as possible
-  std::string hdr(header() + "\n");
-  if (multiAppLevel() > 0)
-    MooseUtils::indentMessage(_name, hdr);
-  Moose::out << hdr << std::flush;
+  auto hdr = header();
+  if (hdr.length() != 0)
+  {
+    if (multiAppLevel() > 0)
+      MooseUtils::indentMessage(_name, hdr);
+    Moose::out << hdr << std::endl;
+  }
```

Comment:
flush was probably here for a reason?

Reply:
The old code has flush as well. I think we want to push the header out on screen asap.